LLMNeedleHaystackTester.py

```diff
@@ -200,70 +165,55 @@ async def evaluate_and_log(self, context_length, depth_percent):
-            response = response.choices[0].message.content
-        elif self.model_provider == "Anthropic":
-            response = await self.model_to_test.completions.create(
-                model=self.model_name,
-                max_tokens_to_sample=300,
-                prompt=prompt,
-                temperature=0
-            )
-            response = response.completion
+        response = await self.get_response_from_model(prompt)
+
 
         test_end_time = time.time()
         test_elapsed_time = test_end_time - test_start_time
 
         # Compare the reponse to the actual needle you placed
         score = self.evaluate_response(response)
 
-
         results = {
             # 'context' : context, # Uncomment this line if you'd like to save the context the model was asked to retrieve from. Warning: This will become very large.
-            'model' : self.model_to_test_description,
-            'context_length' : int(context_length),
-            'depth_percent' : float(depth_percent),
-            'version' : self.results_version,
-            'needle' : self.needle,
-            'model_response' : response,
-            'score' : score,
-            'test_duration_seconds' : test_elapsed_time,
-            'test_timestamp_utc' : datetime.now(timezone.utc).strftime('%Y-%m-%d %H:%M:%S%z')
+            'model': self.model_to_test_description,
+            'context_length': int(context_length),
+            'depth_percent': float(depth_percent),
+            'version': self.results_version,
+            'needle': self.needle,
+            'model_response': response,
+            'score': score,
+            'test_duration_seconds': test_elapsed_time,
+            'test_timestamp_utc': datetime.now(timezone.utc).strftime('%Y-%m-%d %H:%M:%S%z')
         }
 
         self.testing_results.append(results)
 
         if self.print_ongoing_status:
-            print (f"-- Test Summary -- ")
-            print (f"Duration: {test_elapsed_time:.1f} seconds")
-            print (f"Context: {context_length} tokens")
-            print (f"Depth: {depth_percent}%")
-            print (f"Score: {score}")
-            print (f"Response: {response}\n")
+            print(f"-- Test Summary -- ")
+            print(f"Duration: {test_elapsed_time:.1f} seconds")
+            print(f"Context: {context_length} tokens")
+            print(f"Depth: {depth_percent}%")
+            print(f"Score: {score}")
+            print(f"Response: {response}\n")
 
-        context_file_location = f'{self.model_name.replace(".", "_")}_len_{context_length}_depth_{int(depth_percent*100)}'
+        context_file_location = f'{self.model_name.replace(".", "_")}_len_{context_length}_depth_{int(depth_percent * 100)}'
 
         if self.save_contexts:
-            results['file_name'] : context_file_location
+            results['file_name']: context_file_location
 
             # Save the context to file for retesting
             if not os.path.exists('contexts'):
```

Comment:
Consider using `pathlib`, code below could be rewritten to something like

```python
contexts_dir = Path('contexts')
contexts_dir.mkdir(parents=True, exist_ok=True)

context_file_path = contexts_dir / f'{context_file_location}_context.txt'
context_file_path.write_text(context)
```

Reply:
Updated the code to use `pathlib` instead of `os.path` in the latest pull request.